tools/update_roles.php

```diff
@@ -0,0 +1,669 @@
+ *     php update_roles.php [role] remove [permission]
+ *
+ * To recalculate and reasign the roles for every user:
+ *     php update_roles.php refresh
+ *
+ * PHP Version 7
+ *
+ * @category Main
+ * @package  Loris
+ * @author   Tara Campbell <tara.campbell@gmail.com>
+ * @license  Loris license
+ * @link     https://www.github.com/aces/Loris
+ */
+
+require_once __DIR__ . "/../vendor/autoload.php";
+require_once __DIR__ . "/generic_includes.php";
+
+$client = new NDB_Client();
+$client->makeCommandLine();
+$client->initialize(__DIR__."/../project/config.xml");
+
+$roles       = getRoles();
+$permissions = getPermissions();
+
+// User did not provide any
+if (count($argv) === 1) {
+    syntaxIncorrect();
+    exit();
+}
+
+// User asking for the list of roles
+if (isset($argv[1]) && $argv[1] === "roles") {
+    if (!empty($roles)) {
+        print "The roles in the database are:\n\n";
+        prettyPrint($roles);
+    } else {
+        print "There are currently no roles in the database.\n";
+    }
+    
+    exit();
+}
+
+// User asking for the list of permissions
+if (isset($argv[1]) && $argv[1] === "permissions") {
+    print "The permissions in the database are:\n\n";
+    prettyPrint($permissions);
+    exit();
+}
+
+// User asking to reassign roles based on the user permissions
+if (isset($argv[1]) && $argv[1] === "refresh") {
+    refresh();
+    exit();
+}
+
+if (isset($argv[1]) && !isRole($argv[1])) {
+    syntaxIncorrect();
+    exit();
+}
+
+$role = $argv[1];
+
+// User asking for the permissions associated with a role
+if (count($argv) === 2) {
+    print "The permissions for the " . $role . " role are :\n\n";
+    prettyPrint(getRolePermissions($role));
+    exit();
+}
+
+// User looking to add a permission to a role
+if (isset($argv[2]) && $argv[2] === "add") {
+    if (isset($argv[3])) {
+        $permission = $argv[3];
+        if (isPermission($permission)) {
+            addPermission($role, $permission);
+            exit();
+        } else {
+            echo $permission . " is not a valid permission.\n";
+            exit();
+        }
+    } else {
+        echo "You must include the permission you would like to add.\n";
+        exit();
+    }
+}
+
+// User looking to remove a permission from a role
+if (isset($argv[2]) && $argv[2] === "remove") {
+    if (isset($argv[3])) {
+        $permission = $argv[3];
+        if (isPermission($permission)) {
+            removePermission($role, $permission);
+            exit();
+        } else {
+            echo $permission . " is not a valid permission.\n";
+            exit();
+        }
+    } else {
+        echo "You must include the permission you would like to remove.\n";
+        exit();
+    }
+}
+
+syntaxIncorrect();
+exit();
+
+/**
+ * SyntaxIncorrect
+ *
+ * Prints a message to users with the argument instructions
+ *
+ * @return nothing
+ */
+function syntaxIncorrect()
+{
+    print "You have not used the correct argument syntax for this script.\n\n".
```

Comment:
String literals can span multiple lines in PHP. You don't need to have those `\n".` at the end. It would be easier to read the code and see how it's formatted with a single multiline string literal.

Reply:
Done